### DISCUSSION TITLE
How to dump CFG for source code files?

### DISCUSSION THREAD

**Discussion by wangzitom12306** [pmd/pmd#4891] [Q&A]
PMD is a very great project. I'm considering using PMD for taint analysis, which requires control flow graphs.

As mentioned in [link](https://github.com/pmd/pmd/blob/86154aa160b47969878c20c0fc1e15c6ddd3a817/docs/pages/pmd/userdocs/extending/ast_dump.md), AST can be dumped from PMD, either using command line ( `pmd ast-dump` ), or in programmatic way (`XmlTreeRenderer`)

I'm wondering is there a similar way to dump the control flow graphs for java/jsp source code files from PMD?

Thank you very much. Any help is greatly appreciated.

**Comment by jsotuyod** [comment]
No, I fear there is no such mechanism in place.

**Reply by wangzitom12306** [reply]
Got it, thank you very much.
